pyoptsparse/pyOpt_optimizer.py

```diff
@@ -358,7 +358,12 @@ def _masterFunc2(self, x, evaluate, writeHist=True):
 
                 self.userObjTime += time.time() - timeA
                 self.userObjCalls += 1
-                # User values stored is immediately
+
+                # Discard zero imaginary components in funcs
+                for key, val in funcs.items():
```

Comment:
Can you explain why this is needed in addition to the casting above (e.g. `con.value = np.real(copy.copy(c))`)

Reply:
The changes in the `Optimization` class only affect the values stored in `optProb.objectives` and `optProb.constraints` . Here `funcs` is coming straight from `objFun` (line 344), so it could be complex.

Reply:
Right that makes sense.